src/api/Services/RoleOperations.cs

```diff
@@ -117,12 +129,32 @@ internal IList<StorageAccountAndContainers> GetAccessibleContainersForPrincipal(
 
 				// Find an existing entry for this storage account in the result set
 				StorageAccountAndContainers fsr = results
-					.SingleOrDefault(x => x.StorageAccountName.Equals(storageAccountName, StringComparison.OrdinalIgnoreCase));
+					.SingleOrDefault(x => x.Account.StorageAccountName.Equals(storageAccountName, StringComparison.OrdinalIgnoreCase));
 
 				// If this is the first time we've encountered this storage account, Set the storage account name property and add to result set
 				if (fsr == null)
 				{
-					fsr = new StorageAccountAndContainers() { StorageAccountName = storageAccountName };
+					// Check for cached storage account properties for the storage account name
+					var val = storageAccountProperties.Value.FirstOrDefault(x => x.StorageAccountName == storageAccountName);
+
+					// Check for the friendly name in the cache. If it doesn't exist, get it from Azure and add it to the cache.
+					var fname = (val is null) ? String.Empty : val.StorageAccountFriendlyName;
+					if (val is null || fname is null)
+					{
+							fname = rgo.GetAccountResourceTagValue(storageAccountName, StorageAccountFriendlyTagNameKey);

```

Comment:
Use `SasConfiguration.StorageAccountFriendlyNameTagName` instead of a local variable.